src/SarifCli.FunctionalTests/TestData/UseAbsolutePathsForNestedFileUriFragments/NestedFileUriFragmentIsRelativeInCodeFlowLocationUri_Expected.sarif

```diff
@@ -0,0 +1,84 @@
+{
+  "$schema": "http://json.schemastore.org/sarif-1.0.0",
+  "version": "1.0.0",
+  "runs": [
+    {
+      "id": "391375f4-09bc-4f6a-b09a-0c2317759901",
+      "tool": {
+        "name": "SarifCli",
+        "fullName": "SarifCli 1.0.0.0",
+        "version": "1.0.0.0",
+        "semanticVersion": "1.0.0",
+        "sarifLoggerVersion": "1.5.27.0",
+        "language": "en-US",
+        "properties": {
+          "Comments": "Command line tool for working with SARIF log files.",
+          "CompanyName": "Microsoft Corporation",
+          "ProductName": "Microsoft SARIF SDK"
+        }
+      },
+      "results": [
+        {
+          "ruleId": "SV0002",
+          "level": "warning",
+          "formattedRuleMessage": {
+            "formatId": "DefaultFormatId",
+            "arguments": [
+              "NestedFileUriFragmentIsRelativeInCodeFlowLocationUri.sarif",
+              "/runs/0/results/0/codeFlows/0/locations/0/physicalLocation/uri",
+              "file:///c:/bin/myApp.appx#js/minified.js"
+            ]
+          },
+          "locations": [
+            {
+              "analysisTarget": {
+                "uri": "file:///G:/Code/sarif-sdk/src/SarifCli.FunctionalTests/TestData/UseAbsolutePathsForNestedFileUriFragments/NestedFileUriFragmentIsRelativeInCodeFlowLocationUri.sarif",
+                "region": {
+                  "startLine": 40,
+                  "startColumn": 70
+                }
+              }
+            }
+          ]
+        }
+      ],
+      "toolNotifications": [
+        {
+          "id": "MSG001.AnalyzingTarget",
+          "physicalLocation": {
+            "uri": "file:///G:/Code/sarif-sdk/src/SarifCli.FunctionalTests/TestData/UseAbsolutePathsForNestedFileUriFragments/NestedFileUriFragmentIsRelativeInCodeFlowLocationUri.sarif"
+          },
+          "message": "Analyzing 'NestedFileUriFragmentIsRelativeInCodeFlowLocationUri.sarif'...",
+          "level": "note",
+          "time": "2016-08-25T22:51:53.317Z"
+        }
+      ],
+      "rules": {
+        "SV0002": {
+          "id": "SV0002",
+          "name": "UseAbsolutePathsForNestedFileUriFragments",
+          "shortDescription": "When specifying the URI of a nested file with a URI fragment, express the fragment as an absolute path.",
+          "fullDescription": "When specifying the URI of a nested file with a URI fragment, express the fragment as an absolute path. That is, begin the with a forward slash, for example \"file:///C:/bin/MyApp.zip#/images/grape.jpg\".",
+          "messageFormats": {
+            "DefaultFormatId": "In \"{0}\", at \"{1}\", the fragment portion of the URI \"{2}\", which specifies the location of a nested file with respect to its parent, does not begin with a forward slash."
```

Comment:
That's actually not what's happening. There are several things to say here:

1. The code actually _is_ using the Visual Studio formatting helper. It calls into the framework's `ConsoleLogger`, which ultimately calls `FormatForVisualStudio`. So the results produced by the validator _are_ in the correct format, and you _can_ navigate to them.

2. The "{0}" here is indeed the file name, repeated. It's there because, as the framework stands today, it _always_ prepends the file name to the list of message arguments. I complained about this in another comment on this code review (please take a look at that comment). This is really a bug in the framework, and I'm going to file a bug and fix it. BinSkim (which relies on this "feature") will need to be updated so that if it wants the file name in the text of every message -- which apparently it does -- it puts it there itself.

3. The "{1}" here is not the line/column location. It is the JSON pointer location, which specifies the logical path through the JSON to the error. The message would read, for example, "In 'foo.sarif', at '/runs/0/results/0/locations/0/analysisTarget/uri', the fragment portion of the URI 'bar#baz'... blah blah". Once I fix the bug in the framework, it will just read "At /runs/0... ".

---
In reply to: [77240794](https://github.com/Microsoft/sarif-sdk/pull/501#discussion_r77240794) [](ancestors = 77240794)